
<!DOCTYPE HTML>
<html lang="en" >
    <head>
        <meta charset="UTF-8">
        <meta content="text/html; charset=utf-8" http-equiv="Content-Type">
        <title>Heatmap, visualize genotypes in matrix Â· SnpHub Tutorial</title>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
        <meta name="description" content="">
        <meta name="generator" content="GitBook 3.2.3">
        
        
        
    
    <link rel="stylesheet" href="../../gitbook/style.css">

    
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-anchor-navigation-ex/style/plugin.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-prism/prism-ghcolors.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-toggle-chapters/toggle.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-splitter/splitter.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-code/plugin.css">
                
            
                
                <link rel="stylesheet" href="../../gitbook/gitbook-plugin-fontsettings/website.css">
                
            
        

    

    
        
    
        
    
        
    
        
    
        
    
        
    

        
    
    
    <meta name="HandheldFriendly" content="true"/>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black">
    <link rel="apple-touch-icon-precomposed" sizes="152x152" href="../../gitbook/images/apple-touch-icon-precomposed-152.png">
    <link rel="shortcut icon" href="../../gitbook/images/favicon.ico" type="image/x-icon">

    
    <link rel="next" href="hapnet.html" />
    
    
    <link rel="prev" href="vartable.html" />
    

    </head>
    <body>
        
<div class="book">
    <div class="book-summary">
        
            
            
                <nav role="navigation">
                


<ul class="summary">
    
    
    
    

    

    
        
        
    
        <li class="chapter " data-level="1" data-path="../../">
            
                <a href="../../">
            
                    
                        <b>1.</b>
                    
                    What is SnpHub
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="2" data-path="overview.html">
            
                <a href="overview.html">
            
                    
                        <b>2.</b>
                    
                    Overview
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3" data-path="channels.html">
            
                <a href="channels.html">
            
                    
                        <b>3.</b>
                    
                    Channels
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="3.1" data-path="sampleinfo.html">
            
                <a href="sampleinfo.html">
            
                    
                        <b>3.1.</b>
                    
                    SampleInfo, a first glance of the dataset
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.2" data-path="vartable.html">
            
                <a href="vartable.html">
            
                    
                        <b>3.2.</b>
                    
                    VarTable, export region-specific variation tables
            
                </a>
            

            
        </li>
    
        <li class="chapter active" data-level="3.3" data-path="heatmap.html">
            
                <a href="heatmap.html">
            
                    
                        <b>3.3.</b>
                    
                    Heatmap, visualize genotypes in matrix
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.4" data-path="hapnet.html">
            
                <a href="hapnet.html">
            
                    
                        <b>3.4.</b>
                    
                    HapNet, construct a haplotype network
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.5" data-path="phylotree.html">
            
                <a href="phylotree.html">
            
                    
                        <b>3.5.</b>
                    
                    PhyloTree, visualize sample distances as a tree
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.6" data-path="hapmap.html">
            
                <a href="hapmap.html">
            
                    
                        <b>3.6.</b>
                    
                    HapMap, visualize the genotypes geographically
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.7" data-path="snpfreq.html">
            
                <a href="snpfreq.html">
            
                    
                        <b>3.7.</b>
                    
                    SnpFreq, visualize SNP annotation with gene tracks
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="3.8" data-path="seqmaker.html">
            
                <a href="seqmaker.html">
            
                    
                        <b>3.8.</b>
                    
                    SeqMaker, create consensus sequence
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="4" data-path="../Setup/administration_guide.html">
            
                <a href="../Setup/administration_guide.html">
            
                    
                        <b>4.</b>
                    
                    Administration guide
            
                </a>
            

            
            <ul class="articles">
                
    
        <li class="chapter " data-level="4.1" data-path="../Setup/quick_deploy.html">
            
                <a href="../Setup/quick_deploy.html">
            
                    
                        <b>4.1.</b>
                    
                    General Setup
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.2" data-path="../Docker/overview.html">
            
                <a href="../Docker/overview.html">
            
                    
                        <b>4.2.</b>
                    
                    Setup with Docker
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.3" data-path="../Setup/configuration.html">
            
                <a href="../Setup/configuration.html">
            
                    
                        <b>4.3.</b>
                    
                    Configuration
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.4" data-path="../Setup/file-formats.html">
            
                <a href="../Setup/file-formats.html">
            
                    
                        <b>4.4.</b>
                    
                    File Formats
            
                </a>
            

            
        </li>
    
        <li class="chapter " data-level="4.5" data-path="../Setup/uisetting.html">
            
                <a href="../Setup/uisetting.html">
            
                    
                        <b>4.5.</b>
                    
                    UI Setting
            
                </a>
            

            
        </li>
    

            </ul>
            
        </li>
    
        <li class="chapter " data-level="5" data-path="../QA/QA.html">
            
                <a href="../QA/QA.html">
            
                    
                        <b>5.</b>
                    
                    Questions and Answers
            
                </a>
            

            
        </li>
    

    

    <li class="divider"></li>

    <li>
        <a href="https://www.gitbook.com" target="blank" class="gitbook-link">
            Published with GitBook
        </a>
    </li>
</ul>


                </nav>
            
        
    </div>

    <div class="book-body">
        
            <div class="body-inner">
                
                    

<div class="book-header" role="navigation">
    

    <!-- Title -->
    <h1>
        <i class="fa fa-circle-o-notch fa-spin"></i>
        <a href="../.." >Heatmap, visualize genotypes in matrix</a>
    </h1>
</div>




                    <div class="page-wrapper" tabindex="-1" role="main">
                        <div class="page-inner">
                            
                                <section class="normal markdown-section">
                                
                                <div id="anchor-navigation-ex-navbar"><i class="fa fa-navicon"></i><ul><li><span class="title-icon "></span><a href="#heatmap-visualize-genotypes-in-matrix"><b>3.3. </b>Heatmap, visualize genotypes in matrix</a></li><ul><li><span class="title-icon "></span><a href="#panel-&#x2460;-on-the-left-provides-several-options-as-following"><b>3.3.1. </b>Panel &#x2460; on the left provides several options as following:</a></li><li><span class="title-icon "></span><a href="#download-options"><b>3.3.2. </b>Download Options</a></li><li><span class="title-icon "></span><a href="#demonstration"><b>3.3.3. </b>Demonstration</a></li></ul></ul></div><a href="#heatmap-visualize-genotypes-in-matrix" id="anchorNavigationExGoTop"><i class="fa fa-arrow-up"></i></a><h1 id="heatmap-visualize-genotypes-in-matrix"><a name="heatmap-visualize-genotypes-in-matrix" class="anchor-navigation-ex-anchor" href="#heatmap-visualize-genotypes-in-matrix"><i class="fa fa-link" aria-hidden="true"></i></a>3.3. Heatmap, visualize genotypes in matrix</h1>
<p>The <strong>heatmap</strong> channel provides a way to visualize genotype variations across multiple samples in an intuitive interface. Through the heatmap, it&apos;s easy to figure out the mutation patterns or haplotypes and variation types.</p>
<p>Different colors are used to represent the <code>homozygous mutations</code> (dark blue), <code>heterozygous mutations</code> (light blue), <code>reference genotypes</code> (grey), and <code>missing data</code> (white). The annotated variation function and gene-based location information of each position are also shown as two heatmap vectors. The annotated variation functions include <code>missense variations</code>, <code>synonymous variations</code>, <code>frameshift_variant</code>, <code>stop_gained/stop_lost</code> et al. That mutations located in <code>genebody</code> and <code>intergenic</code> regions are also distinguished by different color in another annotation track. Additionally, the chromosome and position are also shown for each variation site.</p>
<p>When the queried samples are in different groups, each group of samples will be displayed in separated block, to help to compare haplotypes between the groups.</p>
<p><img src="../img/Heatmap-1.jpg" alt="Heatmap channel"></p>
<h2 id="panel-&#x2460;-on-the-left-provides-several-options-as-following"><a name="panel-&#x2460;-on-the-left-provides-several-options-as-following" class="anchor-navigation-ex-anchor" href="#panel-&#x2460;-on-the-left-provides-several-options-as-following"><i class="fa fa-link" aria-hidden="true"></i></a>3.3.1. Panel &#x2460; on the left provides several options as following:</h2>
<ul>
<li><p><strong>Groups</strong>: The textbox titled &quot;Groups&quot; inquiries a list of group IDs. Both <code>pre-defined group</code> and <code>user-defined group</code> styles are supported. For more details, please see <a href="https://esctrionsit.github.io/snphub_tutorial/content/Basic_Usage/overview.html" target="_blank">the overview section</a>.</p>
</li>
<li><p><strong>Region or GeneID</strong>: The textbox titled &quot;<strong>Region or GeneID</strong> inquire the input for querying genomic regions. The input text shall be in form of <code>chr:from-to</code>. For example, <code>chr1A:1-100</code>. Also, gene name is acceptable.</p>
</li>
<li><p><strong>Flanking region length</strong>: When using the gene name as input, you may want to extend the regions. It is usually <code>0</code> bp in the box by default, indicating no flanking region is considered. If <code>2000</code> is provided in this input box, then the flanking regions in length of &quot;2000bp&quot; will be included for both upstream and downstream.</p>
</li>
<li><p><strong>Cluster samples according to their genotypes</strong>: The two options <code>Yes</code> and <code>No</code> represent &quot;cluster samples within each group according to their genotype similarity&quot; and &quot;not to cluster samples within each group according to their genotype similarity&quot;.</p>
</li>
<li><p><strong>Flip x and y axis</strong>: By default, genomic positions are displayed in rows, and samples are displayed in columns. The option <code>Yes</code> indicates &quot;to transform the two dimensions&quot;.</p>
</li>
<li><p><strong>Minimum allele frequency (MAF)</strong>: Minor allele frequency (MAF) is the frequency at which the second most common allele occurs in a given population. Higher MAF threshold will result in fewer sites left. If the value is <code>0</code> (by default), this filter function is disabled, and no position will be filtered by MAF.</p>
</li>
<li><p><strong>Draw</strong>: Click for action when getting all options ready.</p>
</li>
</ul>
<h2 id="download-options"><a name="download-options" class="anchor-navigation-ex-anchor" href="#download-options"><i class="fa fa-link" aria-hidden="true"></i></a>3.3.2. Download Options</h2>
<p><img src="../img/Download-options.gif" alt="Download options of Heatmap"></p>
<p>Click the <strong>Download Options</strong> button, and then you can select downloading format (<code>PNG</code> or <code>PDF</code>) of figure, and specify the width and height to appropriate presentation.</p>
<p>Then, click <code>Download</code> to download the figure.</p>
<h2 id="demonstration"><a name="demonstration" class="anchor-navigation-ex-anchor" href="#demonstration"><i class="fa fa-link" aria-hidden="true"></i></a>3.3.3. Demonstration</h2>
<p><img src="../img/Heatmap-0.gif" alt="Demonstration of Heatmap"></p>

                                
                                </section>
                            
                        </div>
                    </div>
                
            </div>

            
                
                <a href="vartable.html" class="navigation navigation-prev " aria-label="Previous page: VarTable, export region-specific variation tables">
                    <i class="fa fa-angle-left"></i>
                </a>
                
                
                <a href="hapnet.html" class="navigation navigation-next " aria-label="Next page: HapNet, construct a haplotype network">
                    <i class="fa fa-angle-right"></i>
                </a>
                
            
        
    </div>

    <script>
        var gitbook = gitbook || [];
        gitbook.push(function() {
            gitbook.page.hasChanged({"page":{"title":"Heatmap, visualize genotypes in matrix","level":"3.3","depth":1,"next":{"title":"HapNet, construct a haplotype network","level":"3.4","depth":1,"path":"content/Basic_Usage/hapnet.md","ref":"./content/Basic_Usage/hapnet.md","articles":[]},"previous":{"title":"VarTable, export region-specific variation tables","level":"3.2","depth":1,"path":"content/Basic_Usage/vartable.md","ref":"./content/Basic_Usage/vartable.md","articles":[]},"dir":"ltr"},"config":{"plugins":["github","-sharing","sharing-plus","anchor-navigation-ex","-highlight","prism","prism-themes","-lunr","-search","expandable-chapters-small","toggle-chapters","splitter","code","livereload"],"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"pluginsConfig":{"prism":{"css":["prism-themes/themes/prism-ghcolors.css"]},"github":{"url":"https://github.com/esctrionsit/snphub"},"livereload":{},"splitter":{},"sharing-plus":{"qq":false,"all":["facebook","google","twitter","instapaper","linkedin","pocket","stumbleupon"],"douban":false,"facebook":true,"weibo":false,"instapaper":false,"whatsapp":false,"hatenaBookmark":false,"twitter":true,"messenger":false,"line":false,"vk":false,"pocket":true,"google":false,"viber":false,"stumbleupon":false,"qzone":false,"linkedin":false},"code":{"copyButtons":true},"fontsettings":{"theme":"white","family":"serif","size":2},"anchor-navigation-ex":{"associatedWithSummary":true,"float":{"floatIcon":"fa fa-navicon","level1Icon":"","level2Icon":"","level3Icon":"","showLevelIcon":false},"mode":"float","multipleH1":false,"pageTop":{"level1Icon":"fa fa-hand-o-right","level2Icon":"fa fa-hand-o-right","level3Icon":"fa fa-hand-o-right","showLevelIcon":false},"printLog":false,"showGoTop":true,"showLevel":true},"prism-themes":{},"expandable-chapters-small":{},"sharing":{"qq":false,"all":["twitter","facebook","google"],"douban":false,"facebook":false,"weibo":false,"instapaper":false,"whatsapp":false,"hatenaBookmark":false,"twitter":false,"messenger":false,"line":false,"vk":false,"pocket":false,"google":false,"viber":false,"stumbleupon":false,"qzone":false,"linkedin":false},"theme-default":{"styles":{"website":"styles/website.css","pdf":"styles/pdf.css","epub":"styles/epub.css","mobi":"styles/mobi.css","ebook":"styles/ebook.css","print":"styles/print.css"},"showLevel":true},"toggle-chapters":{}},"theme":"default","author":"","pdf":{"pageNumbers":true,"fontSize":12,"fontFamily":"Arial","paperSize":"a4","chapterMark":"pagebreak","pageBreaksBefore":"/","margin":{"right":62,"left":62,"top":56,"bottom":56}},"structure":{"langs":"LANGS.md","readme":"README.md","glossary":"GLOSSARY.md","summary":"SUMMARY.md"},"variables":{},"title":"SnpHub Tutorial","language":"en","links":{"sidebar":{}},"gitbook":"3.2.3"},"file":{"path":"content/Basic_Usage/heatmap.md","mtime":"2020-04-04T06:29:06.036Z","type":"markdown"},"gitbook":{"version":"3.2.3","time":"2020-04-04T06:31:15.692Z"},"basePath":"../..","book":{"language":""}});
        });
    </script>
</div>

        
    <script src="../../gitbook/gitbook.js"></script>
    <script src="../../gitbook/theme.js"></script>
    
        
        <script src="../../gitbook/gitbook-plugin-github/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-sharing-plus/buttons.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-expandable-chapters-small/expandable-chapters-small.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-toggle-chapters/toggle.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-splitter/splitter.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-code/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-livereload/plugin.js"></script>
        
    
        
        <script src="../../gitbook/gitbook-plugin-fontsettings/fontsettings.js"></script>
        
    

    </body>
</html>

